transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessengerTransport.kt

```diff
@@ -1,67 +1,60 @@
 package com.genesys.cloud.messenger.transport.core
 
-import com.genesys.cloud.messenger.transport.network.DEFAULT_PING_INTERVAL_IN_SECONDS
+import com.genesys.cloud.messenger.transport.network.*
 import com.genesys.cloud.messenger.transport.network.DeploymentConfigUseCase
 import com.genesys.cloud.messenger.transport.network.PlatformSocket
 import com.genesys.cloud.messenger.transport.network.ReconnectionHandlerImpl
 import com.genesys.cloud.messenger.transport.network.WebMessagingApi
 import com.genesys.cloud.messenger.transport.shyrka.receive.DeploymentConfig
 import com.genesys.cloud.messenger.transport.util.DefaultTokenStore
+import com.genesys.cloud.messenger.transport.util.TokenStore
 import com.genesys.cloud.messenger.transport.util.logs.Log
 import com.genesys.cloud.messenger.transport.util.logs.LogTag
 
 /**
- * Client entry point for the Mobile Messenger Transport SDK.
+ * The entry point to the services provided by the transport SDK.
  */
-object MobileMessenger {
+class MessengerTransport(private val configuration: Configuration, private val tokenStore: TokenStore) {
+
+    constructor(configuration: Configuration, storeKey: String) : this(configuration, DefaultTokenStore(storeKey))
+
+    constructor(configuration: Configuration) : this(configuration, "com.genesys.cloud.messenger")
 
     /**
      * Creates an instance of [MessagingClient] based on the provided configuration.
-     *
-     * @param configuration the configuration parameters for setting up the client.
      */
-    fun createMessagingClient(
-        configuration: Configuration,
-    ): MessagingClient {
+    fun createMessagingClient(): MessagingClient {
         val log = Log(configuration.logging, LogTag.MESSAGING_CLIENT)
         val api = WebMessagingApi(configuration)
         val webSocket = PlatformSocket(log.withTag(LogTag.WEBSOCKET), configuration.webSocketUrl, DEFAULT_PING_INTERVAL_IN_SECONDS)
-        val token = DefaultTokenStore(configuration.tokenStoreKey).token
```

Comment:
Minor. I think it is better to fetch `val token = tokenStore.token`  once and then pass it around instead of opening sharedprefs/vault multiple times as every `tokenStore.token` does.

Reply:
Valid.